statsmodels/kernel_methods/_kde1d_cyclic.py

```diff
@@ -0,0 +1,230 @@
+"""
+This module implements the Cyclic1D KDE estimation method, using FFT to speed up computation on grids.
+"""
+from __future__ import division, absolute_import, print_function
+
+import numpy as np
+from .kde_utils import numpy_trans1d_method, finite
+from ._kde1d_methods import KDE1DMethod, fftdensity, fftdensity_from_binned
+
+
+class Cyclic1D(KDE1DMethod):
+    r"""
+    This method assumes cyclic boundary conditions and works only for closed
+    boundaries.
+
+    The estimation is done with a modified kernel given by:
+
+    .. math::
+
+        \hat{K}(x; X, h, L, U) \triangleq K(z)
+        + K\left(z - \frac{U-L}{h}\right)
+        + K\left(z + \frac{U-L}{h}\right)
+
+    where:
+
+    .. math::
+
+        z = \frac{x-X}{h}
+
+    When computing grids, if the bandwidth is constant, the result is computing
+    using FFT.
+    """
+
+    #: Name of the method, for presentation purposes
+    name = 'cyclic1d'
+
+    def __init__(self):
+        super(Cyclic1D, self).__init__()
+
+    @property
+    def bin_type(self):
+        """
+        Type of the bins adapted for the method
+        """
+        return 'C'
+
+    @numpy_trans1d_method()
+    def pdf(self, points, out):
+        """
+        Compute the PDF of the estimated distribution.
+
+        Parameters
+        ----------
+        points: ndarray
+            Points to evaluate the distribution on
+        out: ndarray
+            Result object. If must have the same shapes as ``points``
+
+        Returns
+        -------
+        out: ndarray
+            Returns the PDF for each point. The default is to use the formula
+            for unbounded pdf computation using the :py:func:`convolve` function.
+        """
+        if not self.bounded:
+            return KDE1DMethod.pdf(self, points, out)
+        if not self.closed:
+            raise ValueError("Cyclic boundary conditions can only be used with "
+                             "closed or un-bounded domains.")
+
+        exog = self.exog
+        points = points[..., np.newaxis]
+
+        # Make sure points are between the bounds
+        if any(points < self.lower) or any(points > self.upper):
+            points = points - self.lower
+            points %= self.upper - self.lower
+            points += self.lower
```

Comment:
Indeed it is. Distribution on angles are a good example (although a simple one where the lower bound is 0). If you want to evaluate the distribution on 7/2 pi, you would expect the same as for pi/2. And for the calculation, I need to get the "reduced" form (i.e. the one between lower and upper).